src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -3078,6 +3081,178 @@ GUI_TEST_CLASS_DEFINITION(test_7659) {
+                              GTGlobals::UseKeyBoard);
+}
+
+GUI_TEST_CLASS_DEFINITION(test_7667) {
+    // Run 2 tasks one by one with different settings.
+
+    // Open _common_data/primer3/only_primer.uql.
+    //     The "Query Designer" window opens.
+    // Run Schema.
+    //     The "Run Schema" dialog appears.
+    // Set:
+    //     Load sequence      _common_data/primer3/all_settingsfiles.fa
+    //     Save results to    tmp/7667.gb
+    // Click Run.
+    //     The all_settingsfiles.fa will open with the resulting annotations.
+    // Expected: the following annotations:
+    //     Result 1    850..869    complement(1022..1041)
+    //     Result 2    22..41      complement(199..218)
+    //     Result 3    850..869    complement(993..1012)
+    //     Result 4    452..471    complement(610..629)
+    //     Result 5    610..629    complement(786..805)
+    // File->Close project. Don't save it.
+    // Return to the "Query Designer" window.
+    // Select the subunit of the Primer element on the Scene.
+    //     Its parameters appear in the Property Editor.
+    // Change the Parameters:
+    //     Product size ranges             100-300
+    //     Number to return                3
+    //     Max repeat mispriming           10
+    //     Max template mispriming         10
+    //     Max 3' stability                10
+    //     Pair max repeat mispriming      20
+    //     Pair max template mispriming    20
+    // Run Schema.
+    //     The "Run Schema" dialog appears.
+    // Click Run.
+    // Expected: the following annotations:
+    //     Result 1    199..218    complement(297..316)
+    //     Result 2    40..59      complement(297..316)
+    //     Result 3    39..58      complement(297..316)
+    GTFileDialog::openFile(os, testDir + "_common_data/primer3/only_primer.uql");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    runSchema(os, testDir + "_common_data/primer3/all_settingsfiles.fa");
+    GTUtilsSequenceView::checkSequenceViewWindowIsActive(os);
+    GTUtilsAnnotationsTreeView::checkAnnotationRegions(os, "Result 1  (0, 2)", {{850, 869}, {1022, 1041}});
+    GTUtilsAnnotationsTreeView::checkAnnotationRegions(os, "Result 2  (0, 2)", {{22, 41}, {199, 218}});
+    GTUtilsAnnotationsTreeView::checkAnnotationRegions(os, "Result 3  (0, 2)", {{850, 869}, {993, 1012}});
+    GTUtilsAnnotationsTreeView::checkAnnotationRegions(os, "Result 4  (0, 2)", {{452, 471}, {610, 629}});
+    GTUtilsAnnotationsTreeView::checkAnnotationRegions(os, "Result 5  (0, 2)", {{610, 629}, {786, 805}});
+    GTUtilsProject::closeProject(os, true);
+    GTUtilsMdi::activateWindow(os, "Query Designer - NewSchema");
+
+    clickOn(GTUtilsQueryDesigner::getItemCenter(os, "Primer"));
+    auto table = GTWidget::findTableView(os, "table");
+    {  // Product size ranges.
+        clickOn(GTTableView::getCellPosition(os, table, 1, 2));
+        GTLineEdit::setText(os, GTWidget::findWidgetByType<QLineEdit*>(os, table, "7667"), "100-300");
```

Comment:
Use `static void setText(GUITestOpStatus& os, const QString& lineEditName...` so it will be 1 method call.

Reply:
The test falls.

Reply:
`GTLineEdit::setText(GUITestOpStatus& os, const QString& lineEditName, ...` uses `GTWidget::findLineEdit` which calls `GTWidget::findExactWidget` which calls `GTWidget::findWidget`. What's the difference between `findWidget` and `findWidgetByType` which I am using? `findWidget` fails if it finds more than 1 widget with the given name. This method also expects a widget name. `findWidgetByType` works the other way around. It is used in `GTUtilsWorkflowDesigner::setCellValue`: https://github.com/ugeneunipro/ugene/blob/86e15e13fc5d7caf1c2806e27e8cad5a8759b44c/src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.cpp#L1098

Reply:
ok